src/clients/entitlements/entitlements.user-scoped.ts

```diff
@@ -62,7 +82,23 @@ export class EntitlementsUserScoped<T extends TEntity = TEntity> {
     }
   }
 
-  async isEntitledToPermission(permissionKey: string): Promise<IsEntitledResult> {
+  private async getFeatureFlagResult(featureKey: string, attributes: Record<string, any>): Promise<IsEntitledResult> {
+    const featureFlags = await this.cache.getFeatureFlags(featureKey);
+
+    for (const flag of featureFlags) {
+      const ffResult = evaluateFeatureFlag(flag, attributes);
+      if (ffResult?.treatment === TreatmentEnum.True) {
```

Comment:
the FF evaluation can return undefined? I think it always returns a value (but worth to double check)